source/app.d

```diff
@@ -89,19 +91,20 @@ class ImageFitness
     }
 }
 
-
-void draw()
+void draw(string input,float mutation = 0.99)
 {
-    ImageFitness fitness = new ImageFitness("mona4.jpg");
-    const uint NUMBER_OF_CIRCLES = 10;
-    geneticAlgorithm!(fitness)(fitness.populationSize * NUMBER_OF_CIRCLES, 0.0, 50, 0.99);
+	ImageFitness fitness = new ImageFitness(input);
+	const uint NUMBER_OF_RECTANGLES = 20;
+	geneticAlgorithm!(fitness)(fitness.populationSize * NUMBER_OF_RECTANGLES, 0.0, 50, mutation);
 }
 
 void main(string[] argv)
 {
-    if (argv.length > 1)
-    {
-        getoptions(argv);
-    }
-    draw();
+	try
+	{
+			auto data = getOptions(argv);
+			draw(data.input, data.probability);
```

Comment:
Nejsem si úplně jistej, jak je tomu v Dčku ale v ostatních jazycích se říká, že by jsi neměl používat výjimky u "performance critical" částí aplikace. Je to z toho důvodu, že přidávají zbytečný overhead, takže bych ho možná pro draw vynechal. Nicméně můžu být totálně wrong, co myslíš @fast4shoot ?